Cargo.toml

```
@@ -50,9 +50,9 @@ codegen_testing = { path = "crates/codegen/testing" }
 infra_cli = { path = "crates/infra/cli" }
 infra_utils = { path = "crates/infra/utils" }
 slang_solidity = { path = "crates/solidity/outputs/cargo/crate" }
```

Not sure about this rename. Our monorepo have many crates, and the list will only keep growing. Having the crate names same as their paths/locations is really helpful to figure out which is which, and makes sure grepping/refactoring things is easy. One look at any dependency in `Cargo.toml`, or a `use` statement in a Rust file and you immediately know if it is internal/external, and where exactly it is located in our complex/deep repository structure.

I was actually thinking about taking this one step further, and renaming `slang_solidity` in the root `Cargo.toml` if we can, for the same reason. `Cargo/crate/Cargo.toml` can stay the same, as the name there affects `cargo publish`. But I'm happy to talk about this later/outside this PR if you prefer so.


```suggestion
solidity_cargo_crate = { package = "slang_solidity", path = "crates/solidity/outputs/cargo/crate" }
```

---

I don't necessarily agree here.

I agree with the fact that it's easier to follow if we mirror crate name <-> directory in the general case, like codegen/parser/runtime bits, as that's straightforward and it's mostly plumbing.

However, the `slang_solidity` is an oddball here, and IMO rightfully so, since it's the main artifact of our pipeline, so I suggest keeping (existing or another) a unique name for it.

The only reason the `slang_solidity_node_addon`/`solidity_npm_crate` exists as a separate crate is to work around the poor Cargo support of cdylib used as rlib/building cdylib; it's basically `slang_solidity` source 1-to-1 but exposed to Cargo as a cdylib-only crate and guaranteed not to be depended on by other packages so that it's not misused.

As such, just mirroring the path as the name is confusing as it shouldn't be really treated like an ordinary crate but rather an unfortunate workaround and just another view of the `slang_solidity` crate.

---

> The only reason the slang_solidity_node_addon/solidity_npm_crate exists as a separate crate is to work around the poor Cargo support of cdylib used as rlib/building cdylib.

To me, that is an implementation detail (the fact that it re-exports `slang_solidity` code, with different build settings). To outside callers, it doesn't really differ much than `pub use slang_solidity::*;`.

> As such, just mirroring the path as the name is confusing as it shouldn't be really treated like an ordinary crate but rather an unfortunate workaround and just another view of the slang_solidity crate.

Because of the above reason, to me, the opposite is more confusing. Right now, it exists next to other crates it is related to (under `outputs/npm`) and has a similar name/prefix, and its name indicates both its location and purpose. The new name does not.

---

But it doesn't re-export `slang_solidity` code nor it has any outside callers and that's the whole point - the only purpose this Cargo package has is to serve as workaround and to output a `.node` artifact for the original `slang_solidity` in the Cargo build pipeline.

I'd say we shouldn't hide the workaround (implementation detail) but make it stick out until we can reasonably fix it in the future (when Cargo improves the cdylib story or napi provides a better build support via e.g. `example` targets as a better workaround).

It's an exceptional crate, so it shouldn't have the same treatment as regular packages, like `codegen_parser_runtime` residing in `codegen/parser/runtime`.

The fact that this crate resides in `outputs/npm/crate` is accidental and shouldn't influence how we name the final packages or artifacts, especially since `solidity_npm_crate` doesn't really tell much if one didn't internalize the repository beforehand, whereas `slang_solidity_node_addon` is pretty self-descriptive, assuming that one knows that `slang_solidity` is the main crate we're building and publishing (which is obvious for Rust consumers atm).

If we want to more closely relate the packages with the file system, maybe we could have `/outputs/cargo/{slang_solidity, slang_solidity_node_addon}`? This would put them together, would allow them to retain the package name and bind two concepts more closely, while letting `/outputs/npm` be about the NPM package itself, consuming the Node addon artifact that's built by a Rust package.

@AntonyBlakey do you have any opinion/do you want to break the tie here? :grinning:

---

`/outputs/cargo/{slang_solidity, slang_solidity_node_addon}` Sounds good to me 👍

---

Done in 340c1985425f31285a436afca7325cfaae172396